sources/editor/Stride.Core.MostRecentlyUsedFiles/MostRecentlyUsedFileCollection.cs

```diff
@@ -61,14 +61,25 @@ public void AddFile(UFile filePath, string strideVersion)
         {
             // Reload settings in case concurrent Game Studio instances are running.
             LoadFromSettings();
-            // Remove it if it was already in the list
-            mostRecentlyUsedFiles.RemoveWhere(x => string.Equals(x.FilePath, filePath, StringComparison.OrdinalIgnoreCase));
-            // Add it on top of the list
-            mostRecentlyUsedFiles.Insert(0, new MostRecentlyUsedFile(filePath) { Version = strideVersion });
-            // Save immediately
+
+            // Create a new MostRecentlyUsedFile instance.
+            var fileToAdd = new MostRecentlyUsedFile(filePath) { Version = strideVersion };
+
+            // Check if the file already exists in the mostRecentlyUsedFiles.
+            if (mostRecentlyUsedFiles.Contains(fileToAdd))
+            {
+                // If it exists, move it to the top of the list.
+                mostRecentlyUsedFiles.MoveMatchingItemToIndex(x => string.Equals(x.FilePath, filePath, StringComparison.OrdinalIgnoreCase), 0);
```

Comment:
This works, however, it will not update the version you last opened the file with.
Say you open project MyGame with Stride 4.1 and then you open the same project with Stride 4.2 and upgrade it. The MRU will still say the version was 4.1.
That's why the version was being removed and then inserted.

I'm not really sure you're fixing it in the right place to be honest. Maybe your repro is incomplete.
have a look at `EditorViewModel.RemoveRecentFile` which is called from `NewOrOpenSessionTemplateCollectionViewModel` and `EditorViewModel.OpenSession`

Reply:
Thanks, @manio143, I will take a look on it.